tests/MongoDB.Driver.Tests/Specifications/unified-test-format/EntityMap.cs

```
@@ -0,0 +1,322 @@
+                    case "client":
+                        if (_clients.ContainsKey(id))
+                        {
+                            throw new Exception($"Client entity with id '{id}' already exists");
+                        }
+                        var createClientResult = CreateClient(entity);
+                        _clients.Add(id, createClientResult.Item1);
+                        _clientEventCapturers.Add(id, createClientResult.Item2);
+                        break;
+                    case "collection":
+                        if (_collections.ContainsKey(id))
+                        {
+                            throw new Exception($"Collection entity with id '{id}' already exists");
+                        }
+                        var collection = CreateCollection(entity);
+                        _collections.Add(id, collection);
+                        break;
+                    case "database":
+                        if (_databases.ContainsKey(id))
+                        {
+                            throw new Exception($"Database entity with id '{id}' already exists");
+                        }
+                        var database = CreateDatabase(entity);
+                        _databases.Add(id, database);
+                        break;
+                    case "session":
+                        if (_sessions.ContainsKey(id))
+                        {
+                            throw new Exception($"Session entity with id '{id}' already exists");
+                        }
+                        var session = CreateSession(entity);
+                        _sessions.Add(id, session);
+                        break;
+                    default:
+                        throw new NotSupportedException($"TODO: '{entityType}'");
+                }
+            }
+        }
+
+        // public methods
+        public IGridFSBucket GetBucket(string bucketId)
+        {
+            return _buckets[bucketId];
```

Should these throw an exception or return null of the the entity is not found?  I suggest throwing (not sure what the behavior of dictionary actually is...

---

Thanks for suggestion!
Dictionary's default is throw, but I think returning `null` is also a good way to go. I don't have a definitive opinion on which is better so I'll leave it as is since it throws on non existing key.